scripts/ping360_socat.py

```diff
@@ -0,0 +1,75 @@
+#!/usr/bin/env python -u
+
+"""
+This manage socat sessions so that every Ping360 device has a screen session
+running socat to expose it to an UDP port.
+"""
+import time
+import subprocess
+import re
+
+def list_ping_devices():
+    """
+    List devices in dev/serial/ping/ping360*
+    """
+    try:
+        # Note: we could easily support Ping1D here.
+        output = subprocess.check_output("ls /dev/serial/ping/Ping360*", shell=True)
+        output = [line for line in output.decode().split("\n") if len(line) > 0]
+        return output
+    except subprocess.CalledProcessError as e:
+        print("Error listing devices at '/dev/serial/ping/ping360*':", e)
+        return []
+
+def device_has_screen(device):
+    """
+    Checks if 'device' has an associated screen session
+    returns true if so
+    """
+    screen_name = device.split("/")[-1].split("-t-")[0]
+    try:
+        output = subprocess.check_output(["sudo", "-Au", "pi", "screen", "-ls"], universal_newlines=True)
+        return output.decode()
+    except subprocess.CalledProcessError as e:
+        if e.returncode == 1:
+            output = e.output.decode() # screen v4.2.x always gives exit code of 1 for 'screen -ls'
+            if not output: # no stdout, just stderr (which we don't capture the contents)
+                print("Error reading screen sessions!")
+
+    regex_string = r"((?P<idNum>[0-9]+)\.(?P<idName>[^\t]+))(\t\(*(?P<time>[0-9][^\)]*)?\))?\t\((?P<state>[A-z]+)\)"
+    regex = re.compile(regex_string)
+    matches = regex.finditer(output)
+
+    sessions = [match.groupdict()['idName'].decode() for match in matches]
+    return screen_name in sessions
+
+def create_device_screen(device, port):
+    """
+    Creates a screen session for 'device' running socat to expose
+    at UDP port 'port'
+    """
+    try:
+        # Follow link to actual device
+        target_device = subprocess.check_output(' '.join(["readlink", "-f", "%s" % device]), shell=True)
+        # Strip newline from output
+        target_device = target_device.decode().split('\n')[0]
+    except Exception as e:
+        print("Error reading symlink: %s" % e)
+        return False
+    screen_name = device.split("/")[-1].split("-t-")[0]
```

Comment:
this is a duplicate of https://github.com/bluerobotics/companion/pull/267/files#diff-cb201b4c22de5793e9f4492c525544ddR29, for a better clarity and to avoid duplicated code, I believe that would be better to create a function for this.